enumerables.rb

```diff
@@ -1,57 +1,53 @@
 require 'pry'
 
-# this method returns an array of hashes, which we'll use in the other methods
-def spicy_foods 
+def spicy_foods
   [
     { name: 'Green Curry', cuisine: 'Thai', heat_level: 9 },
     { name: 'Buffalo Wings', cuisine: 'American', heat_level: 3 },
     { name: 'Mapo Tofu', cuisine: 'Sichuan', heat_level: 6 }
   ]
 end
 
-# given an array of spicy foods, **return an array of strings**
-# with the names of each spicy food
 def get_names(spicy_foods)
-  # your code here
+  spicy_foods.map do |food|
+    food[:name]
+  end
 end
 
-# given an array of spicy foods, **return an array of hashes** 
-# where the heat level of the food is greater than 5
 def spiciest_foods(spicy_foods)
-  # your code here
+  spicy_foods.filter do |food|
+    food[:heat_level] > 5
+  end
 end
 
-# given an array of spicy foods, **output to the terminal**
-# each spicy food in the following format: 
-# Buffalo Wings (American) | Heat Level: 🌶🌶🌶
-# HINT: you can use * with a string to produce the correct number of 🌶 emoji. 
-# "hello" * 3 == "hellohellohello"
 def print_spicy_foods(spicy_foods)
-  # your code here
+  spicy_foods.each do |food|
+    heat = "🌶" * food[:heat_level]
+    puts "#{food[:name]} (#{food[:cuisine]}) | Heat Level: #{heat}"
+  end
 end
 
-# given an array of spicy foods and a string representing a cuisine, **return a single hash**  
-# for the spicy food whose cuisine matches the cuisine being passed to the method
 def get_spicy_food_by_cuisine(spicy_foods, cuisine)
-  # your code here
+  spicy_foods.find do |food|
+    food[:cuisine] == cuisine
+  end
 end
 
-# Given an array of spicy foods, **return an array of hashes** 
-# sorted by heat level from lowest to highest
 def sort_by_heat(spicy_foods)
-  # your code here
+  sorted_array = spicy_foods.sort do |food1, food2|
+    food1[:heat_level] <=> food2[:heat_level]
+  end
 end
 
-# given an array of spicy foods, output to the terminal ONLY 
-# the spicy foods that have a heat level greater than 5, in the following format: 
-# Buffalo Wings (American) | Heat Level: 🌶🌶🌶
-# HINT: Try to use methods you've already written to solve this!
 def print_spiciest_foods(spicy_foods)
-  # your code here
+  filter_foods = spiciest_foods(spicy_foods)
+  print_spicy_foods(filter_foods)
 end
 
-# given an array of spicy foods, return an integer representing 
-# the average heat level of all the spicy foods in the array
 def average_heat_level(spicy_foods)
-  # your code here
+  heat_levels = spicy_foods.map do |food|
+   food[:heat_level]
+  end
+    average_heat = heat_levels.sum / heat_levels.length
+    average_heat
```

Comment:
Another way this could be done. Ruby is great!
```suggestion
  total_heat = spicy_foods.sum do |food|
   food[:heat_level]
  end
  
  total_heat / heat_levels.length
```